client/client.go

```
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"../api"
+	"encoding/json"
+	"flag"
+	"html/template"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+// Errors : to display errors from the server
+type Errors struct {
+	Errs []string
+}
+
+var (
+	mainTmpl    = template.Must(template.ParseFiles("client/mainPage.html"))
+	retNameTmpl = template.Must(template.ParseFiles("client/returnNamePage.html"))
+	errorTmpl   = template.Must(template.ParseFiles("client/errorPage.html"))
+)
+
+var (
+	urlAPI   = flag.String("api", "http://127.0.0.1:8098", "api url")
+	httpAddr = flag.String("addr", ":8111", "server addr")
+)
+
+func main() {
+	flag.Parse()
+
+	http.HandleFunc("/get", handleGet)
+	http.HandleFunc("/set", handleSet)
+	http.HandleFunc("/post/set", handlePostSet)
+	http.HandleFunc("/post/get", handlePostGet)
+
+	log.Println("Start work at :8111...")
+	err := http.ListenAndServe(*httpAddr, nil)
+	if err != nil {
+		log.Fatal("Error happend", err.Error())
+	}
+}
+
+func handleSet(w http.ResponseWriter, r *http.Request) {
+	mainTmpl.Execute(w, "")
+}
+
+func defaultAPI() api.API {
+	return api.NewAPI(*urlAPI)
+}
+
+func handlePostSet(w http.ResponseWriter, r *http.Request) {
+	a := defaultAPI()
+	resp, err := a.Set(r.FormValue("phone"), r.FormValue("name"))
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		errResp, _ := json.Marshal("Error happend: " + err.Error())
+		w.Write(errResp)
```

I assume here is error ignored
Please never ignore errors